control-plane/csi-driver/src/bin/node/node.rs

```diff
@@ -73,6 +77,32 @@ impl Node {
             .chain(vec![self.node_name_segment()])
             .collect()
     }
+
+    /// Get the target URI for the given volume.
+    /// If the REST volume client is enabled, use it to fetch the volume and its target URI.
+    /// Otherwise, use the publish context.
+    async fn volume_uri(
+        &self,
+        volume_id: &Uuid,
+        context: &HashMap<String, String>,
+    ) -> Result<String, tonic::Status> {
+        let ctx_uri = context.get("uri").ok_or_else(|| {
+            failure!(
+                Code::InvalidArgument,
+                "Failed to stage volume {volume_id}: URI attribute missing from publish context"
+            )
+        })?;
+
+        let Some(client) = &self.vol_client else {
+            return Ok(ctx_uri.to_string());
+        };
+
+        let uri = client.volume_uri(volume_id).await?;
+        if &uri != ctx_uri {
+            tracing::warn!(volume.uri=%uri, ctx.uri=%ctx_uri, "Overriding URI volume context with URI from REST");
+        }
+        Ok(uri)
```

Comment:
After we override, the staging will succeed but volumeattachment will still keep holding old uri?

Reply:
Yes it'll still be there

Reply:
> Yes it'll still be there

If the target moves back to old node again at some point in future, with a different uri(say due to transport change or ip/interface change) does the volumeattachment get updated with new publish_context/uri? or the volume attachment is deleted and recreated so it picks up correct new uri?

Reply:
The VA attach data is immutable, it never gets updated :(
If the application itself moves, then we go controller unpublish/publish and then we get a new VA which might have correct uri for some time again

Reply:
ok. so IIUC, for scenario I described - when app never moves and if target comes back again to original node, the original VA will still be there with stale uri, which we'll override with this fix. But without this new option enabled, the staging will fail in this scenario?

Reply:
Yes correct. User can manually get around it by deleting the VA.